tls/s2n_tls13_handshake.c

```diff
@@ -177,3 +177,57 @@ int s2n_tls13_handle_application_secrets(struct s2n_connection *conn)
 
     return 0;
 }
+
+int s2n_update_application_traffic_keys(struct s2n_connection *conn, s2n_mode mode, keyupdate_status status)
+{
+    notnull_check(conn);
+    
+    /* get tls13 key context */
+    s2n_tls13_connection_keys(keys, conn);
+
+    struct s2n_session_key *old_key;
+    struct s2n_blob old_app_secret;
+    struct s2n_blob sequence_number;
+
+    if (mode == S2N_CLIENT) {
+        old_key = &conn->secure.client_key;
+        GUARD(s2n_blob_init(&old_app_secret, conn->secure.client_app_secret, keys.size));
+        sequence_number.data = conn->secure.client_sequence_number;
+        sequence_number.size = sizeof(conn->secure.client_sequence_number);
+    } else {
+        old_key = &conn->secure.server_key;
+        GUARD(s2n_blob_init(&old_app_secret, conn->secure.server_app_secret, keys.size));
+        sequence_number.data = conn->secure.server_sequence_number;
+        sequence_number.size = sizeof(conn->secure.server_sequence_number);
+    }
+
+    /* Produce new application secret */
+    s2n_stack_blob(app_secret_update, keys.size, S2N_TLS13_SECRET_MAX_LEN);
+
+    /* Derives next generation of traffic secret */
+    GUARD(s2n_tls13_update_application_traffic_secret(&keys, &old_app_secret, &app_secret_update));
+
+    s2n_tls13_key_blob(app_key, conn->secure.cipher_suite->record_alg->cipher->key_material_size);
+    struct s2n_blob app_iv = { .data = conn->secure.client_implicit_iv, .size = S2N_TLS13_FIXED_IV_LEN };
+
+    /* Derives next generation of traffic key */
+    GUARD(s2n_tls13_derive_traffic_keys(&keys, &app_secret_update, &app_key, &app_iv));
+    if (status) {
```

Comment:
nit: would be clearer with status == RECEIVING

Reply:
I'll get this on the keyupdate send PR :)